app/controllers/api/base_controller.rb

```diff
@@ -4,5 +4,11 @@ module Api
   class BaseController < ActionController::API
     include DeviseTokenAuth::Concerns::SetUserByToken
     before_action :authenticate_user!, unless: :devise_controller?
+
+    rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+
+    def record_not_found
+      render json: { error: 'User not found.' }, status: :not_found
```

Comment:
In the future this could be a post or another model, so you should have a different message just in case (like record not found for example)